oauthproxy.go

```diff
@@ -1130,12 +1138,12 @@ func validOptionalPort(port string) bool {
 // - `nil, ErrAccessDenied` if the authenticated user is not authorized
 // Set-Cookie headers may be set on the response as a side-effect of calling this method.
 func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.Request) (*sessionsapi.SessionState, error) {
-	var session *sessionsapi.SessionState
+	session := middlewareapi.GetRequestScope(req).Session
 
-	getSession := p.sessionChain.Then(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		session = middlewareapi.GetRequestScope(req).Session
-	}))
-	getSession.ServeHTTP(rw, req)
+	// Check this after loading the session so that if a valid session exists, we can add headers from it
+	if p.IsAllowedRequest(req) {
```

Comment:
So I understand the motivation here to get headers from sessions into passthrough allowed paths.

But is this this the right place to do the `IsAllowedRequest` check in the nesting? Now it is applied to more endpoints than just `Proxy` (userinfo, auth) and the logic for this allowlist codepath is a little obfuscated.

Is there a way to make it its own explicit method decoupled from `getAuthenticatedSession` to keep the concerns separate?

Reply:
Also, I can't remember if I touched on this in the original Issue reporting:

Is injecting in headers if a session happens to exist for an allowed request a breaking change. I've run into scenarios in the past where users would allowlist a path and then inject their explicit override for a specific header for that route.

If we have header middleware, we'd override or strip those for security purposes right?

I'll also need to think about session vs no session allowed paths and how header security works a bit more.

Reply:
> But is this this the right place to do the IsAllowedRequest check in the nesting? Now it is applied to more endpoints than just Proxy (userinfo, auth) and the logic for this allowlist codepath is a little obfuscated.

At this point, `getAuthenticatedSession` acts as our authorization logic (should probably rename it). So this logic fits with the rest of the content of this method at least. I think once we get our authz layer sorted, this function will go away completely.

The rest of your questions are all valid but I haven't had a chance to really look into or think about this now. It's been so long since I implemented this initially that I can't remember what's going on

Reply:
> But is this this the right place to do the IsAllowedRequest check in the nesting? Now it is applied to more endpoints than just Proxy (userinfo, auth) and the logic for this allowlist codepath is a little obfuscated.

Previously, the `IsAllowedRequest` was really early in the switch statement. So if you accessed `Userinfo` or `Auth`, and were an allowed request, you actually got directed back to the `Proxy`. Ergo, if you had a valid SkipAuth and a valid session, there was no way to access userinfo or the auth endpoint. So I think with this change, we are fixing a bug. We aren't applying the `IsAllowedRequest` to more endpoints as it previously trumped these other endpoints anyway

Reply:
> Is injecting in headers if a session happens to exist for an allowed request a breaking change. I've run into scenarios in the past where users would allowlist a path and then inject their explicit override for a specific header for that route.
>
> If we have header middleware, we'd override or strip those for security purposes right?

We already have the headers chain in the path for anything that skips the auth route anyway, so it would have stripped anything that a user tried to inject that matched any header we would set. Ergo I don't see this as a breaking change. We are simply adding extra info where before there was none. If the user had turned off the stripping, this may be a change but we have discussed security implications of this before and I think it's up to the user how they want to deal with this (they'll probably already be doing so for when it's not an allowed route)